torchvision/transforms/functional_tensor.py

```diff
@@ -914,7 +902,26 @@ def equalize(img: Tensor) -> Tensor:
 
     _assert_channels(img, [1, 3])
 
-    if img.ndim == 3:
-        return _equalize_single_image(img)
+    # img is a ...xCxHxW tensor
+    # temp is a ...xCxHxWx256 tensor
+    temp = img[..., None] == torch.arange(0, 256, dtype=torch.uint8, device=img.device)
+    height_dim, width_dim = -3, -2
+    hist = temp.sum(dim=[height_dim, width_dim])
+
+    step = (hist.sum(dim=-1) - hist[..., -1]) // 255
+    step = step[..., None]
+    if (step == 0).all():
+        return img
 
-    return torch.stack([_equalize_single_image(x) for x in img])
+    luts = (torch.cumsum(hist, dim=-1) + (step // 2)) // step
+    luts = torch.nn.functional.pad(luts, [1, 0])[..., :-1].clamp(0, 255)
+    # luts (lookup tables) is a ...xCx265 tensor
+
+    img = img.to(torch.int64)
+    if img.ndim == 3:
+        return _remap_single_image(img, luts)
+    else:  # more than one image
+        imgs = img
+        return torch.stack([
+            _remap_single_image(img, luts) for (img, luts) in zip(imgs, luts)
+        ])
```

Comment:
Ideally we would get rid of the `stack()` calls: at that point all we want to do is to remap `img (...xCxHxW)` according to the values in `luts (...xCx256)`.

There's probably a vetorized one-liner to do that, maybe with `gather()`, but I couldn't get it to work. Any help welcome :)

Reply:
You can get both the single image and the batch of image to work with the approach I mentioned just above. The implementation would look somewhat like
```python
idx_batch = torch.arange(0, batch_size, device=imgs.device)[:, None, None, None]
idx_channels = torch.arange(0, num_channels, device=imgs.device)[None, :, None, None]
return luts[idx_batch, idx_channels, imgs]
```